### DISCUSSION TITLE
Ability to disable saving of some exact chunks

### DISCUSSION THREAD

**Discussion by Dymeth** [PaperMC/Paper#10499] [Ideas]
### Is your feature request related to a problem?

There is currently no any ways to prevent individual chunks from being saved. This applies to both regular automatic incremental saving and API-based saving.

### Describe the solution you'd like.

It would be very convenient to have a `ChunkSaveEvent` with the ability to cancel this event, or some other mechanism to control saving of chunks. Note that this event should also be called while unloading worlds too.

In addition, a `World#saveChunk(Chunk)` or `World#saveChunkAsync(Chunk)` method would be useful.

At this moment we have `World#unloadChunk()` with the `save` parameter, but it is practically useless because either no chunk is saved or all chunks are saved during the game before unloading.

### Describe alternatives you've considered.

The only alternative at the moment is to disable chunk saving completely, and unload chunks with the `save` parameter when shutting down the plugin/server.
But this is an extremely unsafe approach, because in case of server problems the data of useful chunks will be lost

**Comment by codebycam** [comment]
Hey Dymeth, 

Is the [`ChunkUnloadEvent#setSaveChunk`](https://jd.papermc.io/paper/1.20/org/bukkit/event/world/ChunkUnloadEvent.html#setSaveChunk(boolean)) what you're looking for?

**Comment by Dymeth** [comment]
> Hey Dymeth,
> 
> Is the [`ChunkUnloadEvent#setSaveChunk`](https://jd.papermc.io/paper/1.20/org/bukkit/event/world/ChunkUnloadEvent.html#setSaveChunk(boolean)) what you're looking for?

In this issue we are talking about saving chunks, not about unloading. Please read the whole top post
